cpp/src/strings/capitalize.cu

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/case.hpp>
+#include <cudf/utilities/error.hpp>
+#include <strings/char_types/is_flags.h>
+#include <strings/utilities.hpp>
+#include <strings/utilities.cuh>
+
+
+namespace cudf
+{
+namespace strings
+{
+namespace detail
+{
+namespace { // anonym.
+  
+  //base class for probing string
+  //manipulation memory load requirements;
+  //
+  struct probe_base
+  {
+    probe_base(column_device_view const d_column,
+               character_flags_table_type const* d_flags,
+               character_cases_table_type const* d_case_table):
+      d_column_(d_column),
+      d_flags_(d_flags),
+      d_case_table_(d_case_table)
+    {  
+    }
+
+    __host__ __device__
+    column_device_view const get_column(void) const
+    {
+      return d_column_;
+    }
+
+    __host__ __device__
+    character_flags_table_type const* get_flags(void) const
+    {
+      return d_flags_;
+    }
+
+    __host__ __device__
+    character_cases_table_type const* get_case_table(void) const
+    {
+      return d_case_table_;
+    }
+
+    __device__
+    virtual int32_t operator()(size_type idx) const = 0;
```

Comment:
Recommend removing this line. Detailed comment in the same method for the other base class.

Reply:
Same counter-argument as above.

Reply:
Done.